internal/concierge/server/server.go

```diff
@@ -270,5 +270,18 @@ func getAggregatedAPIServerScheme(apiGroup string) *runtime.Scheme {
 	utilruntime.Must(loginv1alpha1.RegisterConversions(scheme))
 	utilruntime.Must(loginv1alpha1.RegisterDefaults(scheme))
 
+	// on incoming requests, restore the authenticator API group to the standard group
+	scheme.AddTypeDefaultingFunc(&loginv1alpha1.TokenCredentialRequest{}, func(obj interface{}) {
+		credentialRequest := obj.(*loginv1alpha1.TokenCredentialRequest)
+		if credentialRequest.Spec.Authenticator.APIGroup == nil {
+			return
+		}
+		restoredGroup, ok := groupsuffix.Unreplace(*credentialRequest.Spec.Authenticator.APIGroup, apiGroupSuffix)
+		if !ok {
```

Comment:
Say that someone is running with a custom API group (e.g., `walrus.tld`), and they `create TokenCredentialRequest` with `TokenCredentialRequest.Spec.Authenticator.APIGroup: "login.concierge.pinniped.dev"`. Then the `credentialRequest.Spec.Authenticator.APIGroup` wouldn't get changed by `Unreplace()`, and so we would perform a cache lookup with `APIGroup: "login.concierge.pinniped.dev"`, which would succeed since we are using that default suffix in the code[0]. But this seems like we are lying to the caller - there really isn't any object with `APIGroup: "login.concierge.pinniped.dev"`, so we should return `"no such authenticator"`.

WDYT?

[0] https://github.com/vmware-tanzu/pinniped/blob/9399b5d8005955e44b4001a8c4928bcff22493c3/internal/controller/authenticator/webhookcachefiller/webhookcachefiller.go#L72

Reply:
I can set it to an invalid group name in this case, WDYT?

Reply:
Resolved in Slack here: https://kubernetes.slack.com/archives/C01BW364RJA/p1612558605224600?thread_ts=1612551663.223500&cid=C01BW364RJA.